pandas/core/strings/object_array.py

```diff
@@ -0,0 +1,423 @@
+import re
+import textwrap
+from typing import Pattern, Set, Union, cast
+import unicodedata
+import warnings
+
+import numpy as np
+
+import pandas._libs.lib as lib
+import pandas._libs.missing as libmissing
+import pandas._libs.ops as libops
+from pandas._typing import Scalar
+
+from pandas.core.dtypes.common import is_re, is_scalar
+from pandas.core.dtypes.missing import isna
+
+from pandas.core.accessor import CachedAccessor
+from pandas.core.arrays.numpy_ import PandasArray
+from pandas.core.strings.base import BaseStringArrayMethods
+
+
+class ObjectArrayMethods(BaseStringArrayMethods):
+    _default_na_value = np.nan
+
+    def _map(self, f, na_value=None, dtype=None):
+        arr = self._array  # object-dtype ndarray.
```

Comment:
this is actually an object-dtype ndarray *wrapped* in ObjectProxy/PandasArray?

Reply:
I'll need to check... My types say it should be an `ExtensionArray` (ObjectProxy is an ExtensionArray subclass). I think we have

```python
Series.str : pandas.core.strings.StringMethods
pandas.core.strings.StringMethods._array : ExtensionArray (specifically, ObjectProxy, Categorical, StringArray)
pandas.core.strings.StringMethods._array._str : BaseStringArrayMethods
pandas.core.strings.StringMethods._array._str._array : ExtensionArray
```

My head is hurting a bit from all the nesting. Surely this can be simplified?

Reply:
> My head is hurting a bit from all the nesting. Surely this can be simplified?

Well, with what I said above: define the methods directly on the ExtensionArray, so you would only have `pandas.core.strings.StringMethods._array.<method>` (without the extra `._str`). But of course that has other downsides.